server/main_test.go

```diff
@@ -0,0 +1,218 @@
+package main_test
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+	"github.com/mattermost/mattermost/server/v8/channels/api4"
+	"github.com/mattermost/mattermost/server/v8/channels/app"
+	"github.com/mattermost/mattermost/server/v8/channels/store/storetest"
+	"github.com/mattermost/mattermost/server/v8/config"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// mainTest is a testing.T-like structure that currently just mimics the t.Cleanup semantics.
+type mainTest struct {
+	cleanupFunctions []func()
+}
+
+// Cleanup adds a function to be called when cleaning up.
+func (mt *mainTest) Cleanup(f func()) {
+	mt.cleanupFunctions = append(mt.cleanupFunctions, f)
+}
+
+// Done calls all cleanup functions with defer-like semantics (last function added called first).
+func (mt *mainTest) Done() {
+	for i := range mt.cleanupFunctions {
+		f := mt.cleanupFunctions[len(mt.cleanupFunctions)-i-1]
+		f()
+	}
+}
+
+// setupServerPath defines MM_SERVER_PATH correctly for server initialization. One day this won't
+// be necessary at all.
```

Comment:
One can only dream...

Reply:
I **will** get back to https://github.com/mattermost/mattermost/pull/24968 one day :)